loader/loader.go

```diff
@@ -398,25 +438,47 @@ func (l *Loader) interpretTagValue(tag string) (*proto.ExtensionDesc, interface{
 	case "github.com/gunk/opt/http.Match":
 		// an error would be caught in Eval
 		expr, _ := parser.ParseExpr(tag)
-		rule := &annotations.HttpRule{}
+
+		// Capture the values required to use in 'annotations.HttpRule'.
+		// We need to evaluate the entire expression, and then we can
+		// create an `annotations.HttpRule'.
+		var path string
+		var body string
+		method := "GET"
 		for _, elt := range expr.(*ast.CompositeLit).Elts {
 			kv := elt.(*ast.KeyValueExpr)
 			val, _ := strconv.Unquote(kv.Value.(*ast.BasicLit).Value)
-			method := "GET"
 			switch name := kv.Key.(*ast.Ident).Name; name {
 			case "Method":
 				method = val
 			case "Path":
-				switch method {
-				case "GET":
-					rule.Pattern = &annotations.HttpRule_Get{Get: val}
-				case "POST":
-					rule.Pattern = &annotations.HttpRule_Post{Post: val}
-				}
+				path = val
+				// TODO: grpc-gateway doesn't allow paths with a trailing '/', should
+				// we return an error here, because the error from grpc-gateway is very
+				// cryptic and unhelpful?
+				// https://github.com/grpc-ecosystem/grpc-gateway/issues/472
 			case "Body":
-				rule.Body = val
+				body = val
 			}
```

Comment:
while at it, make the default here error too

Reply:
Added.